aodndata/soop/soop_ba.py

```diff
@@ -0,0 +1,119 @@
+                    non_nc.publish_type = PipelineFilePublishType.ARCHIVE_ONLY
+                    dest_archive = SoopBaFileClassifier.archive_path(nc)
+                    non_nc.archive_path = os.path.join(dest_archive, non_nc.name)
+                else:
+                    non_nc.publish_type = PipelineFilePublishType.UPLOAD_ONLY
+                    non_nc.dest_path = os.path.join(destination, non_nc.name)
+                    if results:
+                        get_previous_version(self, results, non_nc.name)
+
+        else:  # singe netcdf file
+            netcdf = self.file_collection.filter_by_attribute_id('file_type', FileType.NETCDF)
+            nc = netcdf[0]
+            nc.check_type = PipelineFileCheckType.NC_COMPLIANCE_CHECK
+            destination = SoopBaFileClassifier.dest_path(nc)
+            nc.dest_path = os.path.join(destination, nc.name)
+
+            #  delete previous version base on filetype/extension
+            # netcdf file have to be deleted since filename contains creation date=>never overwritten
+            results = self.state_query.query_storage(destination)
+            if results:
+                get_previous_version(self, results, nc.name, destination)
+
+
+    dest_path = SoopBaFileClassifier.dest_path
+    archive_path = SoopBaFileClassifier.archive_path
+
+
+def get_previous_version(self, previous_file_list, input_file):
+    """
+        Find previous version of each incoming file based on its type/extension and
+        add them to the filecollection with the correct publish type
+        extension can be: .inf', '.nc.png','.pitch.csv','.roll.csv',.gps.csv'
+        inputs: previous_file_list : dictionary containing file listing(full path) and metadata from destination
+              input_file :  file basename
+              destination : full destination path
+    """
+    keylist = previous_file_list.keys()
+    if input_file.endswith('.nc'):
+        r = re.compile(".*.nc$")
+        prev_file = filter(r.match, keylist)
+        assert len(prev_file) == 1, "Found more than one previous versions of the netcdf file. Aborting "
+        print prev_file[0]
+        add_to_collection(self, prev_file[0], PipelineFilePublishType.DELETE_UNHARVEST)
+    else:
+        # check previous file widcard :
+        # can be '.inf', '.nc.png','.pitch.csv','.roll.csv',.gps.csv'
+        previous_version_extension = input_file.split('.')[1:]
+        p = previous_version_extension
+        if len(p) > 1:
+            extension = "%s%s%s%s%s" % ('.*.', p[0], '.', p[1], '$')
+        else:
+            extension = "%s%s%s" % ('.*.', p[0], '$')
+
+        r = re.compile(extension)
+        prev_file = filter(r.match, keylist)
+        assert len(prev_file) == 0, 'Found more than one previous versions of the extension %s. Aborting' % extension
+        add_to_collection(self, prev_file[0], PipelineFilePublishType.DELETE_ONLY)
+
+
+def add_to_collection(self, file, publish_type):
```

Comment:
same as above. didn't look in details, but the unittests are passing, so must be something weird going on